src/cephadm/cephadm

```diff
@@ -2197,6 +2197,9 @@ def list_daemons(detail=True, legacy_dir=None):
     if legacy_dir is not None:
         data_dir = os.path.abspath(legacy_dir + data_dir)
 
+    # keep track of ceph versions we see
+    seen_versions = {}
```

Comment:
cephadm:2201: error: Need type annotation for 'seen_versions' (hint: "seen_versions: Dict[<type>, <type>] = ...")

```suggestion
    seen_versions = {}  # type: Dict[str, Optional[str]]
```